Content.Shared/Xenoarchaeology/Artifact/SharedXenoArtifactSystem.Graph.cs

```diff
@@ -0,0 +1,451 @@
+using System.Diagnostics.CodeAnalysis;
+using Content.Shared.Xenoarchaeology.Artifact.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Xenoarchaeology.Artifact;
+
+/// <summary>
+/// User-friendly API for viewing and modifying the complex graph relationship in XenoArtifacts
+/// </summary>
+public abstract partial class SharedXenoArtifactSystem
+{
+    /// <summary>
+    /// Gets the index corresponding to a given node, throwing if the node is not present.
+    /// </summary>
+    public int GetIndex(Entity<XenoArtifactComponent> ent, EntityUid node)
+    {
+        if (TryGetIndex((ent, ent), node, out var index))
+        {
+            return index.Value;
+        }
+
+        throw new ArgumentException($"node {ToPrettyString(node)} is not present in {ToPrettyString(ent)}");
+    }
+
+    public bool TryGetIndex(Entity<XenoArtifactComponent?> ent, EntityUid node, [NotNullWhen(true)] out int? index)
```

Comment:
Document public API

Reply:
done, all of shared methods received loads of xml-doc